tests/integration/test_referral.cairo

```diff
@@ -0,0 +1,57 @@
+%lang starknet
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from src.interface.starknetid import StarknetId
+from src.interface.naming import Naming
+
+@external
+func __setup__() {
+    %{
+        from starkware.starknet.compiler.compile import get_selector_from_name
+        context.starknet_id_contract = deploy_contract("./lib/starknetid/src/StarknetId.cairo").contract_address
+        context.pricing_contract = deploy_contract("./src/pricing/main.cairo", [123]).contract_address
+        logic_contract_class_hash = declare("./src/naming/main.cairo").class_hash
+        context.naming_contract = deploy_contract("./lib/cairo_contracts/src/openzeppelin/upgrades/presets/Proxy.cairo", [logic_contract_class_hash,
+            get_selector_from_name("initializer"), 4, 
+            context.starknet_id_contract, context.pricing_contract, 456, 0]).contract_address
+    %}
+    return ();
+}
+
+@external
+func test_simple_buy_without_sponsor{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    alloc_locals;
+    local starknet_id_contract;
+    local naming_contract;
+    %{
+        ids.starknet_id_contract = context.starknet_id_contract
+        ids.naming_contract = context.naming_contract
+        stop_prank_callable1 = start_prank(456, context.starknet_id_contract)
+        stop_prank_callable2 = start_prank(456, context.naming_contract)
+        stop_mock_1 = mock_call(123, "transferFrom", [1])
+        stop_mock_2 = mock_call(10101010, "add_commission", [1])
+        warp(1, context.naming_contract)
+    %}
+
+    // Set referral contract address
+    Naming.set_referral_contract(naming_contract, 10101010);
+
+    let token_id = 1;
+    StarknetId.mint(starknet_id_contract, token_id);
+    // th0rgal encoded
+    let th0rgal_string = 28235132438; // th0rgal encoded
+
+    Naming.buy(naming_contract, token_id, th0rgal_string, 365, 0, 456, 0, new (0));
+    let (addr) = Naming.domain_to_address(naming_contract, 1, new (th0rgal_string));
+    assert addr = 456;
+
+    Naming.buy(naming_contract, 2, 11111111, 365, 0, 456, 1, new (th0rgal_string));
```

Comment:
Why do you buy with a sponsor now ?

Reply:
Because we updated the buy function to accept a sponsor as argument, and calling it without the arg sponsor wouldn't work. Other possibility would be to create a new function like `buy_sponsored`